src/collection-facets/more-facets-content.ts

```diff
@@ -45,59 +48,91 @@ import {
 } from '../utils/analytics-events';
 import './toggle-switch';
 import { srOnlyStyle } from '../styles/sr-only';
+import {
+  mergeSelectedFacets,
+  sortBucketsBySelectionState,
+  updateSelectedFacetBucket,
+} from '../utils/facet-utils';
 
 @customElement('more-facets-content')
 export class MoreFacetsContent extends LitElement {
   @property({ type: String }) facetKey?: FacetOption;
 
-  @property({ type: String }) facetAggregationKey?: FacetOption;
-
   @property({ type: String }) query?: string;
 
   @property({ type: Object }) filterMap?: FilterMap;
 
-  @property({ type: Object }) modalManager?: ModalManagerInterface;
-
-  @property({ type: Object }) searchService?: SearchServiceInterface;
-
   @property({ type: String }) searchType?: SearchType;
 
   @property({ type: Object }) pageSpecifierParams?: PageSpecifierParams;
 
   @property({ type: Object })
   collectionTitles?: CollectionTitles;
 
+  /**
+   * Maximum number of facets to show per page within the modal.
+   */
+  @property({ type: Number }) facetsPerPage = 35;
+
+  /**
+   * Whether we are waiting for facet data to load.
+   * We begin with this set to true so that we show an initial loading indicator.
+   */
+  @property({ type: Boolean }) facetsLoading = true;
+
+  /**
+   * The set of pre-existing facet selections (including both selected & negated facets).
+   */
   @property({ type: Object }) selectedFacets?: SelectedFacets;
 
   @property({ type: String }) sortedBy: AggregationSortType =
     AggregationSortType.COUNT;
 
-  @property({ type: Object, attribute: false })
-  analyticsHandler?: AnalyticsManagerInterface;
-
-  @state() aggregations?: Record<string, Aggregation>;
+  @property({ type: Object }) modalManager?: ModalManagerInterface;
 
-  @state() facetGroup?: FacetGroup[] = [];
+  @property({ type: Object }) searchService?: SearchServiceInterface;
 
-  @state() facetGroupTitle?: String = '';
+  @property({ type: Object, attribute: false })
+  analyticsHandler?: AnalyticsManagerInterface;
 
-  @state() pageNumber = 1;
+  /**
+   * The full set of aggregations received from the search service
+   */
+  @state() private aggregations?: Record<string, Aggregation>;
 
   /**
-   * Facets are loading on popup
+   * A FacetGroup storing the set of facet buckets shown on the current page.
    */
-  @state() facetsLoading = true;
+  @state() private facetGroup?: FacetGroup;
 
-  @state() paginationSize = 0;
+  /**
+   * An object holding any changes the patron has made to their facet selections
+   * within the modal dialog but which they have not yet applied. These are
+   * eventually merged into the existing `selectedFacets` when the patron applies
+   * their changes, or discarded if they cancel/close the dialog.
+   */
+  @state() private unappliedFacetChanges = getDefaultSelectedFacets();
```

Comment:
New facet selections within the modal are saved here across different pages of facets, and only merge them in later when the patron chooses to apply them.